presto-main/src/main/java/com/facebook/presto/sql/planner/SubqueryPlanner.java

```diff
@@ -318,10 +319,69 @@ private PlanBuilder appendQuantifiedComparisonApplyNode(PlanBuilder subPlan, Qua
                         return appendQuantifiedComparisonApplyNode(subPlan, rewrittenAny, correlationAllowed);
                 }
                 break;
+
+            case LESS_THAN:
+            case LESS_THAN_OR_EQUAL:
+            case GREATER_THAN:
+            case GREATER_THAN_OR_EQUAL:
+                // A < ALL B <=> A < min B
+                // A > ALL B <=> A > max B
+                // A < ANY B <=> A < max B
+                // A > ANY B <=> A > min B
+                QualifiedName aggregationFunction = chooseAggregationFunction(quantifiedComparison);
+                RelationPlan subqueryRelationPlan = createRelationPlan(quantifiedComparison.getSubquery());
+                PlanNode subqueryPlan = subqueryRelationPlan.getRoot();
+                Symbol outputColumn = getOnlyElement(subqueryRelationPlan.getOutputSymbols());
+                Type outputColumnType = getOnlyElement(subqueryRelationPlan.getDescriptor().getAllFields()).getType();
+                checkState(outputColumnType.isOrderable(), "Subquery result type must be orderable");
+                FunctionRegistry functionRegistry = metadata.getFunctionRegistry();
+                Symbol subValue = symbolAllocator.newSymbol(aggregationFunction.toString(), outputColumnType);
+                subqueryPlan = new AggregationNode(
```

Comment:
I think you can add a commit where you extract creation aggregation node with min max aggreagtions for the subquery as separate method. Which could be the used in several rewrites for quantified comparisons.